syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.syndeo.transforms.datagenerator;
+
+import com.google.common.flogger.GoogleLogger;
+import java.io.Serializable;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import org.apache.avro.Schema;
+import org.apache.avro.Schema.Parser;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.utils.AvroUtils;
+import org.apache.beam.sdk.transforms.FlatMapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.PeriodicImpulse;
+import org.apache.beam.sdk.transforms.Reshuffle;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class DataGeneratorSchemaTransform implements SchemaTransform, Serializable {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private long recordsPerSecond = 1000l;
+  private long minutesToRun = 10;
+  private String schema;
```

Comment:
these can all be `private final` variables

Reply:
better to keep them not final, since class members suppose to be changeable.